packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -61,7 +61,7 @@ function BlockBindingsPanelDropdown( { fieldsList, attribute, binding } ) {
 								{ registeredSources[ name ].label }
 							</DropdownMenuV2.GroupLabel>
 						) }
-						{ Object.entries( fields ).map( ( [ key, value ] ) => (
+						{ Object.entries( fields ).map( ( [ key, args ] ) => (
 							<DropdownMenuV2.RadioItem
 								key={ key }
 								onChange={ () =>
```

Comment:
This part needs further enhancements to become more flexible. In particular:

```js
[ attribute ]: {
	source: name,
		args: { key },
}
```

In my opinion, every source should construct `args` argument, which for Post Meta happens to be `{ key }`. For Pattern Overrides that would be `{}` or `undefined`, and every other source could decide what that it.

In effect, I believe the object item from the `fields` should have the following shape:

```js
{
    label: string,
    value: string,
    args: any,
}
```

This way it's up to the implementor to provide all that's necessary to make it work when defining `getFieldsList` fir the source.

Reply:
I plan to create an issue to discuss the whole `getFieldsList` API to ensure it feels good before making it public.

I believe I shouldn't have used the `args` variable name here because it doesn't refer to the source args, but the object returned by `getFieldsList`. There are two different things:

* The binding object: It specifies which attribute is connected to which source, which can already use any `args` they want. For example, in post meta we use `key` to specify which custom field the attribute is connected to:
```
"content": {
	source: "core/post-meta",
		args: { key: "my_custom_field" },
}
```

* The `getFieldsList` API: This is used just to get a list of ALL the bindable fields in order to show them in the UI and let the user select. In this case, I believe we only need the `label` and the `value`. As it is a callback, each source can decide what to show as the label and the value.

I hope we can clarify it by improving the `getFieldsList` API (probably needs a new name) and that part of the code, because I agree it feels confusing right now.